### ISSUE TITLE
s3.listObjects() data variable returns null

### ISSUE THREAD

**Issue by roberthelmick08** [aws/aws-sdk-js#1833]
I'm trying to write my first lambda function using the Serverless framework that will copy files from the ````texts/```` folder to the ````target/```` folder in an S3 bucket called ````s3-trigger-serverless-test````. When calling the ````listObjects()```` function though, I receive the following error:

    TypeError: Cannot read property 'Contents' of null
        at Response.<anonymous> (/var/task/src/handler.js:14:12)
        at Request.<anonymous> (/var/task/node_modules/aws-sdk/lib/request.js:364:18)
        at Request.callListeners (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:105:20)
        at Request.emit (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:77:10)
        at Request.emit (/var/task/node_modules/aws-sdk/lib/request.js:683:14)
        at Request.transition (/var/task/node_modules/aws-sdk/lib/request.js:22:10)
        at AcceptorStateMachine.runTo (/var/task/node_modules/aws-sdk/lib/state_machine.js:14:12)
        at /var/task/node_modules/aws-sdk/lib/state_machine.js:26:10
        at Request.<anonymous> (/var/task/node_modules/aws-sdk/lib/request.js:38:9)
        at Request.<anonymous> (/var/task/node_modules/aws-sdk/lib/request.js:685:12)

handler.js:

    var AWS = require('aws-sdk');
    var bucketName = 's3-trigger-serverless-test';
    var oldPrefix = 'texts/';
    var s3 = new AWS.S3({params: {Bucket: bucketName}});

    module.exports.copyFilesToS3 = (event, context, callback) =>  {
	    s3.listObjects({Bucket: bucketName, Prefix: oldPrefix}, function(err, data) {
	        if (data.Contents.length) {           // this line results in the error above

serverless.yml:

    functions:
      copyFilesToS3:
        handler: src/handler.copyFilesToS3
        events:
          - s3: 
              bucket: s3-trigger-serverless-test
              event: s3:ObjectCreated:*
              rules:
                - prefix: 'texts/'

The aws-sdk version is up-to-date, as suggested by another stackoverflow answer. I'm thinking it might be because the Bucket and Prefix parameters I passed into ````listObjects()```` didn't return any data from the S3 bucket, but I've played around with other parameters and bucket settings, and I haven't been able to come up with a solution yet.

Thanks in advance

**Comment by jeskew** [comment]
Hi @roberthelmick08,

When `data` is `null`, does the `err` parameter to your callback contain a value? Callbacks will always be called with either `err` populated and `data` set to `null` or `err` set to `null` and `data` populated.

**Comment by jeskew** [comment]
I'm not an IAM expert, but does using `arn:aws:iam::my_12_digit_account_number:root` mean that the permission is only extended to the account's root user? My understanding is that Serverless [will always run lambda functions using an IAM role](https://serverless.com/framework/docs/providers/aws/guide/iam/).

**Comment by fubar** [comment]
@roberthelmick08 I just realized that the SDK returns an Access Denied error when trying to list an empty bucket. Might be a bug, but I'm just starting to dig deeper. Is the bucket or folder you're trying to list empty too?

Also note that changes to a bucket (ie. adding or removing an item) might take a few moments to be reflected through the SDK. I've added an item to my empty bucket and it took a minute or two to show up through the SDK.
